src/dune_engine/action.ml

```diff
@@ -300,6 +303,7 @@ let is_useful_to memoize =
     | System _ -> true
     | Bash _ -> true
     | Extension (module A) -> A.Spec.is_useful_to ~memoize
+    | Needed_deps _ -> memoize
```

Comment:
What does `memoize` mean here?

Reply:
It means that it makes sense to store the rule in the cache, but it does not make sense to look up the target of the action in the distributed (or shared) cache. However, giving a second look at our final definition of `needed_deps`, I think that both of these operations are not necessary, so this should be set to `false`.